backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CAdapterGenerator.kt

```diff
@@ -838,8 +841,8 @@ internal class CAdapterGenerator(val context: Context) : DeclarationDescriptorVi
         output("typedef float              ${prefix}_KFloat;")
         output("typedef double             ${prefix}_KDouble;")
         output("typedef void*              ${prefix}_KNativePtr;")
-        output("struct ${prefix}_KType;")
-        output("typedef struct ${prefix}_KType ${prefix}_KType;")
+        output("struct KType;")
```

Comment:
Why so?

Reply:
Just because same reason as with `KObjHeader*` - it's an opaque reference type.

Reply:
`KObjHeader*` seems to be private for generated `.cpp` file, but `KType` is user-visible. C compiler indeed can handle multiple such `KType` declarations, but these don't represent the same type, so I'm not sure if it's ok to merge them.

Reply:
Yeah, may make sense, as indeed this way it's easier to mix types from two potentially different runtimes, if more than one library is used.